pkgs/applications/blockchains/polkadot/default.nix

```
@@ -31,48 +32,41 @@ rustPlatform.buildRustPackage rec {
     '';
   };
 
+  preBuild = ''
+    export SUBSTRATE_CLI_GIT_COMMIT_HASH=$(cat .git_commit)
```

Bash has built-in support for this: https://www.gnu.org/software/bash/manual/html_node/Command-Substitution.html

```suggestion
    export SUBSTRATE_CLI_GIT_COMMIT_HASH=$(< .git_commit)
```